app/helpers/validation_helper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+function validate_between($str, $min, $max)
+{
+    $length = mb_strlen($str); // count characters based on encoding
+    return $length <= $max && $length >= $min;
+}
+
+function validate_username($str)
+{
+    return preg_match('/^[[:alnum:]_]*$/', $str);
+}
+
+function validate_unique_name($username)
+{
+    return !User::getByUsername($username);
```

Comment:
I think that username should be defined unique in the table.
It will not affect the validation since you call `validate()` first before inserting to the table.
Please check the example below on how to check pdo exceptions when there's duplicate entry
```php
# user.php
if ($this->validate()) {
    throw new ValidationException('Invalid input!');
}

$params = array(
    ...
);

try {
    $db->insert('user', $params);
} catch (PDOException $e) {
    // duplicate entry, do something else
    if ($e->errorInfo[1] == 1062) {
        throw new DuplicateEntryException;
    }
}
```